src/MiniEstablishments/Market/marketEventData.ts

```diff
@@ -0,0 +1,201 @@
+        return `People around start to yell and shout as they realize ${lib.articles.output(lib.random(animal))} has escaped from its cage. The animal is going wild through the market as ${lib.articles.output(lib.random(chase))} chases after it.`
+      }
+    },
+    brawl: {
+      function (town) {
+        const npc1 = createNPC(town)
+        const npc2 = createNPC(town)
+        return `A crowd has gathered near the center of the market to watch a brawl that has broken out. ${profile(npc1, lib.articles.output(npc1.descriptor).toUpperFirst())} is in the center of the crowd currently holding ${profile(npc2, lib.articles.output(npc2.descriptor))} in a headlock and shouting something that can't be made out.`
+      }
+    },
+    music: {
+      function (town) {
+        const npc = createNPC(town, {
+          hasClass: true,
+          dndClass: 'bard'
+        })
+        const instrument = ['lute', 'pair of drumes', 'harp', 'flute', 'pan flute', 'sitar', 'fiddle', 'citern', 'vielle', 'clavichord', 'harpsichord']
+        const playing = ['rather well', 'rather poorly', 'very loudly', 'quite softly', 'and it sounds pretty good', 'and it sounds bad', 'and it sounds alright']
+        return `${profile(npc, lib.articles.output('bard').toUpperFirst())} is standing in a corner of the market playing ${lib.articles.output(lib.random(instrument))} ${lib.random(playing)}. Every once in a while a passerby throws a coin into a cap sitting in front of the bard.`
+      }
+    },
+    snakeCharmer: {
+      exclusions (town) {
+        return !town.bans.includes('animals')
+      },
+      function (town) {
+        const npc = createNPC(town, {
+          profession: 'snake charmer'
+        })
+        return `${profile(npc, lib.articles.output('snake charmer').toUpperFirst())} is sitting on a small rug in the market playing a strange looking flute. In front of the snake charmer is a basket with a large cobra in it hypnotically swaying from side to side. Every so often an onlooker drops some coins into a bag in front of the snake.`
+      }
+    },
+    magicalWares: {
+      function (town) {
+        const npc = createNPC(town, {
+          profession: 'merchant'
+        })
+        const magic = ['firing off miniature fireworks into the air', 'making items disappear into a ring', 'pulling items from seemingly thin air', 'casting a few weak cantrips', 'juggling a few brightly glowing orbs', 'modeling a cloak that looks as if it were wreathed in fire', 'drinking from a mug that is seemingly bottomless', 'pulling impossibly large things from a rather small bag']
+        return `${profile(npc, lib.articles.output(npc.descriptor).toUpperFirst())} is showing off ${npc.hisher} magical wares by ${lib.random(magic)} to the amazement of a small crowd that has gathered around the stall.`
```

Comment:
You could tie the reaction to the `wageVariation` or `professionLuck` properties.